src/components/Accounts.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import type { KeyringPair } from '@polkadot/keyring/types';
+import React from 'react';
+import { Container, Dropdown } from 'semantic-ui-react';
+import styled from 'styled-components';
+
+import AccountActions from '../actions/accountActions';
+import { useAccountContext, useUpdateAccountContext } from '../contexts/AccountContextProvider';
+import { useSourceTarget } from '../contexts/SourceTargetContextProvider';
+import useAccounts from '../hooks/useAccounts';
+import useDerivedAccount from '../hooks/useDerivedAccount';
+
+interface Props {
+  className?: string;
+}
+
+const formatOptions = (accounts: Array<KeyringPair>) =>
+  accounts.map(({ meta, address }) => ({
+    icon: 'user',
+    key: address,
+    text: (meta.name as string).toLocaleUpperCase(),
+    value: address
+  }));
+
+const Accounts = ({ className }: Props) => {
+  const accounts = useAccounts();
+  const derivedAccount = useDerivedAccount();
+  const { dispatchAccount } = useUpdateAccountContext();
+  const { sourceChain, targetChain } = useSourceTarget();
+  const { account: currentAccount } = useAccountContext();
```

Comment:
I'm not a fan of 4 seemingly separate hooks that are basically super tightly coupled. Why can't we simply have:
```
const { accounts, setCurrentAccount, currentAccount, derivedAccount } = useAccounts();
```
?

If you don't want to bundle all things together then let's at least make this similar to the state hooks and merge getter and a function to alter the state.
```
const [account, setAccount] = useCurrentAccount();
```

Reply:
ok, that would add another hook to collect them all , but i see your point.

Reply:
But why another hook? This is account-related funcitonality, I think it can easily stay together in one file without getting overwhelmingly big? But even one more hook is better than too fine-grained and low-level API expose to high-level components IMHO.